Rename and relocate Draw Minimap editor property

Cykyrios opened the pull request:
This is more consistent with other properties, as per #31631.
Show Minimap is now in the Open Scripts tab as it is not technically related to line numbers.

*Bugsquad edit:* Fixes #31631.

akien-mga commented:
There are quite a few other settings in the `line_numbers` category which don't have much to do with line numbers, this could be worth cleaning up.

Not sure the API should be changed, for the editor options "Show Minimap" makes sense and is consistent with other options, but from an API perspective, it's meaningful to make clear that it's not just about "showing" something at no cost, but enables drawing something that can impact performance.

Cykyrios commented:
> Not sure the API should be changed, for the editor options "Show Minimap" makes sense and is consistent with other options, but from an API perspective, it's meaningful to make clear that it's not just about "showing" something at no cost, but enables drawing something that can impact performance.

Should I keep "draw" internally and only change the name to "Show" in the editor settings then?

> There are quite a few other settings in the `line_numbers` category which don't have much to do with line numbers, this could be worth cleaning up.

I must say I'm not too sure which settings should be moved, gutters are not strictly related to line numbers but are displayed next to them, code folding and word wrap make sense as they can cause line numbers to change, and that leaves the line length guildeline, which is indeed unrelated to line numbers (unless it could automatically trigger word wrap, which it doesn't).

Paulb23 commented:
> Should I keep "draw" internally and only change the name to "Show" in the editor settings then?

That sounds fine to me, for all intensive purposes the minimap implementation follows the exact same pattern as line numbers, breakpoints etc, they all use "draw" internally. 

As for where it belongs, I think renaming the category to something more general then "Line Numbers" such as "Appearance" / "Display"? should be okay.

Cykyrios commented:
I reverted to simply renaming the editor option "Show Minimap", and had another look at the "Line Numbers" and "Open Scripts" categories. I would suggest that Open Scripts only contain settings about the open scripts list (which I believe is its original purpose), so settings related to scroll could go into the Appearance category, which would be the new name for the Line Numbers category.

akien-mga commented:
"Appearance" seems consistent with what the Kate editor uses for some of those settings:
![Screenshot_20190830_091928](https://user-images.githubusercontent.com/4701338/64001045-45eed380-cb07-11e9-8cba-5973de09e35f.png)

They also have "Editing Options" for some other settings which overlap with ours.
![Screenshot_20190830_092311](https://user-images.githubusercontent.com/4701338/64001288-cf060a80-cb07-11e9-87d1-7f7213698aa5.png)

The current proposal is already a clear improvement so we could merge as is. I feel that there's still room for improvement though.
One idea could be to have a "Navigation" category for the scrolling and minimap settings.

akien-mga commented:
"Open scripts" could maybe be renamed to "Script list" to reduce the confusion, and make clear that it's about the left-side panel listing scripts and optional bottom section with a list of member functions of the selected script?

Cykyrios commented:
I agree on both points, this makes things clearer.
I just have one problem, after making the changes, I have some leftovers: vertical scroll is still in the Appearance category, and all but one setting from the Open Scripts category is still visible in said category, even though I renamed the category to Script List (so I have duplicate entries). I manually removed these entries from my editor settings file, but I don't know how to handle these from code, any idea?

akien-mga commented:
Looks good to me, could you squash commits together?

>  I just have one problem, after making the changes, I have some leftovers: vertical scroll is still in the Appearance category, and all but one setting from the Open Scripts category is still visible in said category, even though I renamed the category to Script List (so I have duplicate entries). I manually removed these entries from my editor settings file, but I don't know how to handle these from code, any idea?

That's normal, previous values are indeed kept for settings even if they are removed from older versions. This is mostly a legacy thing, but it also enables using the same editor config for different releases even if setting names changed.

akien-mga commented on editor/code_editor.cpp:
```diff
@@ -894,26 +894,26 @@ void CodeTextEditor::update_editor_settings() {
-	text_editor->set_draw_minimap(EditorSettings::get_singleton()->get("text_editor/line_numbers/draw_minimap"));
-	text_editor->set_minimap_width(EditorSettings::get_singleton()->get("text_editor/line_numbers/minimap_width"));
-	text_editor->set_draw_info_gutter(EditorSettings::get_singleton()->get("text_editor/line_numbers/show_info_gutter"));
+	text_editor->set_bookmark_gutter_enabled(EditorSettings::get_singleton()->get("text_editor/appearance/show_bookmark_gutter"));
```
You could maybe group them together by category here, following the same order as in the `editor_settings.cpp` definition. That might make the list easier to read.

Cykyrios replied:
Agreed, I just grouped and reordered them per category and per setting to match what you can see in the editor settings menu. I also squashed the commits.

Cykyrios committed:
Reorganize & rename text editor settings categories

Add Navigation category for scroll and minimap settings.
Rename Line Numbers category to Appearance.
Rename Open Scripts category to Script List.
Rename "Draw Minimap" setting to "Show Minimap" (this is more consistent with other settings).

Reorder settings by category in code_editor.cpp to match settings list

akien-mga commented:
Thanks!
